prusti-viper/src/encoder/builtin_encoder.rs

```diff
@@ -69,6 +77,11 @@ impl BuiltinEncoder {
             BuiltinFunctionKind::Undefined(vir::Type::Bool) => format!("builtin$undef_bool"),
             BuiltinFunctionKind::Undefined(vir::Type::TypedRef(_)) => format!("builtin$undef_ref"),
             BuiltinFunctionKind::Undefined(vir::Type::Domain(_)) => format!("builtin$undef_doman"),
+            BuiltinFunctionKind::ArrayLookupPure { array_ty, array_len, .. } => {
+                // we kind of abuse the TypedRef variant here to store the rust type name as string
```

Comment:
This is true of a lot of Prusti in general though, no need to comment on it here :)

Reply:
removed.